app/presenters/tree_node/menu/node.rb

```diff
@@ -0,0 +1,39 @@
+module TreeNode
+  module Menu
+    class Node < TreeNode::Node
+      set_attribute(:no_click, true)
+      set_attribute(:icon, "pficon pficon-folder-close")
+
+      set_attribute(:checkable) { @options[:editable] }
+
+      def self_and_child_states
+        child_states.unshift(selected)
+      end
+
+      private
+
+      def child_states
+        @states ||= children.flat_map(&:self_and_child_states)
+      end
+
+      def select_by_children_selected
+        return true        if all_children_fully_selected?
+        return 'undefined' if any_children_selected?
+
```

Comment:
@hayesr While I understand that this is more of a matter of opinion, I'm afraid it looks like you're the only one having this opinion (as you may already have noticed, during multiple discussions with me, @skateman, @martinpovolny and most recently @isimluk), so I'd respectfully ask you do ask asked regardless of your personal preference. 

```ruby
def select_by_children_selected
  return false if children.empty?
  return true if child_states.all? { |s| s == true } # true not just truthy
  'undefined' # partially checked
end
```

is all we're asking for, and since this involves reading and understanding 1 method instead of 3, on 3 lines instead of 7, I'm sure you can manage to read this ;).